keps/sig-apps/1847-autoremove-statefulset-pvcs/README.md

```diff
@@ -0,0 +1,544 @@
+      1. Create 2 pod statefulset, delete stateful set, confirm PVCs deleted
+      1. Create 2 pod statefulset, add data to PVs, manually delete one pod, confirm pod comes back and PV has data (PVC not deleted).
+      1. As above, but manually delete all pods in stateful set.
+      1. Create 2 pod statefulset, add data to PVs, manually delete one pod, immediately scale down to one pod, confirm PVC exists.
+      1. Create 2 pod statefulset, add data to PVs, manually delete one pod, immediately scale down to one pod, scale back to two pods, confirm PV has data.
+    - Retain: 
+      1. same tests as above, but PVCs not deleted in any case and confirm data intact on the PV.
+    - Pod restart tests:
+      1. Create statefulset, perform rolling update 
+    - `--casecade=false` tests.
+      1. Manual pod deletion
+      1. The DeleteOnScaleDown tests
+      1. The StatefulSetDeletion tests.
+    - TODO: some of these tests may be suitable for unit tests.
+1. Upgrade/Downgrade tests
+    1. Create statefulset in previous version and upgrade to the version 
+       supporting this feature. The PVCs should remain intact.
+    2. Downgrade to earlier version and check the PVCs with Retain
+       remain intact and the others with set policies before upgrade 
+       gets deleted based on if the references were already set.
+1. Feature disablement/enable test for alpha feature flag `statefulset-autodelete-pvcs`.
+
+
+### Graduation Criteria
+
+#### Alpha release
+- Complete adding the items in the 'Changes required' section.
+- Add unit, functional, upgrade and downgrade tests to automated k8s test.
+
+### Upgrade / Downgrade Strategy
+
+This features adds a new field to the StatefulSet. The default value for the new field
+maintains the existing behavior of StatefulSets.
+
+On a downgrade, the `PersistentVolumeClaimReclaimPolicy` field will be hidden on
+any StatefulSets. The behavior in this case will be identical to mutating they
+policy field to `Retain`, as described above, including the edge cases
+introduced if this is done during a scale-down or StatefulSet deletion.
+
+### Version Skew Strategy
+There are only kubecontroller manager changes involved (in addition to the
+apiserver changes for dealing with the new StatefulSet field). Node components
+are not involved so there is no version skew between nodes and the control plane.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: StatefulSetAutoDeletePVC
+    - Components depending on the feature gate
+      - kube-controller-manager, which orchestrates the volume deletion.
+      - kube-apiserver, to manage the new policy field in the StatefulSet
+        resource (eg [dropDisabledFields](https://github.com/kubernetes/kubernetes/blob/97d40890d00acf721ecabb8c9a6fec3b3234b74b/pkg/api/pod/util.go#L433)).
```

Comment:
nit: I would remove the link - it's misleading, because it is for a pod.
Just mentioning dropDisabledFields is enough